agent/custom/general_navigation_action.py

```
@@ -488,24 +798,33 @@ def run(self, context: Context, _argv: CustomAction.RunArg) -> CustomAction.RunR
 
-                # 5c. 到底判定: 比较连续两屏检测框像素坐标, 完全一致才判定到底
-                # (关卡名集合相同但坐标不同 = 地图实际在移动, 不算到底)
+                # 5c. 到底判定: 看坐标表 Y 最大(底部)的关卡, 连续两屏屏幕y变化不大 => 地图已到底
                 if cur_quests:
                     empty_count = 0   # 检测到关卡 -> 重置连续空屏计数
-                    if last_quests is not None and cur_quests == last_quests:
-                        mfaalog.info(f"[导航] 滑到底, 关卡集合 {cur_set} 与上屏坐标完全一致")
-                        break
+                    if coords:
+                        bot_name, bot_sy, bot_ty = None, -1, -1
+                        for name, nx, ny in cur_quests:
```

Ruff (0.16.1)

[warning] Loop control variable `nx` not used within loop body

(B007)